openshift/dynamic/apply.py

```diff
@@ -71,8 +76,6 @@ def get_deletions(last_applied, desired):
         desired_value = desired.get(k)
         if desired_value is None:
             patch[k] = None
-        elif type(last_applied_value) != type(desired_value):
```

Comment:
Hmm, can we safely remove this without changing other behavior? We could also add an or condition that checks if both types are members of six.string_types rather than removing out entirely

Reply:
This change would only affect when `last_applied_value` is a dict and `desired_value` is not a dict. This would be a major issue if desired_value is not a dict. So perhaps just change line 76 to check both last_applied_value and desired_value are dicts.

Reply:
Changing that also allows us to get rid of the `if desired_value is None` check.